preview/MsixCore/msixmgr/WideString.hpp

```diff
@@ -0,0 +1,647 @@
+#pragma once
+
+#include <assert.h>
+
+#ifndef APPMODEL_WIDESTRING_H_
+#define APPMODEL_WIDESTRING_H_
+#define COMMON_STRING_EMPTY { 0, NULL }
+
+#include <IntSafe.h>
+
+namespace MsixCoreLib
+{
+    struct COMMON_STRING
+    {
+        ULONG length;
+        __field_nullterminated __field_ecount(length + 1) WCHAR* chars;
+    };
+
+    struct CONST_STRING
+    {
+        ULONG length;
+        __field_nullterminated __field_ecount(length + 1) const WCHAR* chars;
+    };
+
+    static const ULONG MaxValue = 0xFFFFFFFF;
+
+    static HRESULT Multiply(
+        _In_ ULONG d1,
+        _In_ ULONG d2,
+        _Out_ ULONG *result)
+    {
+        (*result) = d1 * d2;
+        if ((d2 == 0 || d1 <= MaxValue / d2))
+        {
+            return NOERROR;
+        }
+        return HRESULT_FROM_WIN32(ERROR_ARITHMETIC_OVERFLOW);
+    }
+
+    static HRESULT AddNew(
+        _In_ ULONG d1,
+        _In_ ULONG d2,
+        __out_range(>= , d1) __out_range(>= , d2) ULONG *result)
+    {
+        (*result) = d1 + d2;
+        if (d1 <= MaxValue - d2)
+        {
+            return NOERROR;
+        }
+        return HRESULT_FROM_WIN32(ERROR_ARITHMETIC_OVERFLOW);
+    }
+
+    class String

```

Comment:
>String [](start = 10, length = 6)

can we use std::wstring instead?

Reply:
Removed the string class and its functions as it was not being used anywhere.